src/ergw_dns.erl

```diff
@@ -0,0 +1,95 @@
+%% Copyright 2018, Travelping GmbH <info@travelping.com>
+
+%% This program is free software; you can redistribute it and/or
+%% modify it under the terms of the GNU General Public License
+%% as published by the Free Software Foundation; either version
+%% 2 of the License, or (at your option) any later version.
+
+%% @doc
+%% This modules implements DNS-based Node Discovery according to 3GPP TS 29.303.
+%% Right now this makes decission by lookup NAPTR record for given name. This 
+%% understands service and protocol names which are described in section 
+%% 19.4.3 of 3GPP TS 23.003. 
+
+-module(ergw_dns).
+
+-export([naptr/1, lookup/1, lookup/2]).
+
+%%====================================================================
+%% API
+%%====================================================================
+
+%% @private
+naptr(Name) ->
+    case application:get_env(ergw, nameservers, []) of
+        [] -> inet_res:resolve(Name, in, naptr);
+        Nameservers ->
+            inet_res:resolve(Name, in, naptr, [{nameservers, Nameservers}])
```

Comment:
@mgumz I would think that we can rely on the DNS config of the host/vm/pod. Do we really need to make the name server configurable?